src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderUtils.ts

```diff
@@ -0,0 +1,174 @@
+            text: 'Not equals',
+            data: {
+                operatorType: 'Simple',
+                operatorSymbol: '!='
+            }
+        }
+    ];
+    switch (propertyType) {
+        case 'string':
+            operators.push({
+                key: '3',
+                text: 'Contains',
+                data: {
+                    operatorType: 'Function',
+                    operatorFunction: (property: string, value: string) => {
+                        return `CONTAINS(T.${property}, ${value})`;
+                    }
+                }
+            });
+            operators.push({
+                key: '4',
+                text: 'Not contains',
+                data: {
+                    operatorType: 'Function',
+                    operatorFunction: (property: string, value: string) => {
+                        return `NOT CONTAINS(T.${property}, ${value})`;
+                    }
+                }
+            });
+            break;
+        case 'float':
+        case 'integer':
+        case 'double':
+        case 'long':
+            operators.push({
+                key: '3',
+                text: 'Greater than',
+                data: {
+                    operatorType: 'Simple',
+                    operatorSymbol: '>'
+                }
+            });
+            operators.push({
+                key: '4',
+                text: 'Less than',
+                data: {
+                    operatorType: 'Simple',
+                    operatorSymbol: '<'
+                }
+            });
+            operators.push({
+                key: '5',
+                text: 'Greater or equal',
+                data: {
+                    operatorType: 'Simple',
+                    operatorSymbol: '>='
+                }
+            });
+            operators.push({
+                key: '6',
+                text: 'Less or equal',
+                data: {
+                    operatorType: 'Simple',
+                    operatorSymbol: '<='
+                }
+            });
+            break;
+        default:
+    }
+    return operators;
+};
+
+export const buildQuery = (querySnippets: QueryRowData[]) => {
+    let fullQuery = `SELECT *\nFROM DIGITALTWINS T\n`;
+    querySnippets.forEach((snippet, index) => {
+        if (snippet.operatorData.operatorType === 'Function') {
+            if (index !== 0) {
+                fullQuery = fullQuery.concat(snippet.combinator);
+            }
+            fullQuery = fullQuery.concat(
+                `WHERE ${snippet.operatorData.operatorFunction(
+                    snippet.property,
+                    snippet.value
+                )}\n`
+            );
+        } else {
+            if (index !== 0) {
+                fullQuery = fullQuery.concat(`${snippet.combinator} `);
+            }
+            fullQuery = fullQuery.concat(
+                `WHERE ${snippet.property} ${snippet.operatorData.operatorSymbol} ${snippet.value}\n`
```

Comment:
yeah, as we discussed, just wanted to highlight that there are two "WHERE" clause appended but it is fixed in the subsequent PR: https://github.com/microsoft/iot-cardboard-js/pull/688